src/iminuit/cost.py

```diff
@@ -1126,12 +1126,12 @@ class BinnedCostWithModel(BinnedCost):
     :meta private:
     """
 
-    __slots__ = "_xe_shape", "_model", "_model_xe"
+    __slots__ = "_xe_shape", "_model", "_model_xe", "_n"
 
     def __init__(self, n, xe, model, verbose):
         """For internal use."""
         self._model = model
-
+        self._n = n
```

Comment:
It is not necessary to add `_n` to the slots. If you follow the inheritance chain upwards to `MaskedClost`, you can see that the data array `n` is stored in the slot `_data`. So you can query `self._data` instead of `self._n`

Reply:
Thanks. Done.